build/write_git_metadata.sh

```diff
@@ -6,14 +6,14 @@ BUILD_METADATA_ENV_FILE=$2
 VARIABLE_NAME=$3
 
 # if passed as 4th argument this is the hash
-if [ -z $4 ]; then
-    REPO_GIT_HASH="$(git -C $REPO_PATH rev-parse HEAD)"
+if [ -z "$4" ]; then
+    REPO_GIT_HASH="$(git -C "$REPO_PATH" rev-parse HEAD)"
 else
     REPO_GIT_HASH=$4
 fi
 
-if [ -f $BUILD_METADATA_ENV_FILE ]; then
-    sed -i "/$VARIABLE_NAME/d" $BUILD_METADATA_ENV_FILE
+if [ -f "$BUILD_METADATA_ENV_FILE" ]; then
+    sed -i "/$VARIABLE_NAME/d" "$BUILD_METADATA_ENV_FILE"
 fi
 
-echo "$VARIABLE_NAME="$REPO_GIT_HASH"" >> $BUILD_METADATA_ENV_FILE
+echo "$VARIABLE_NAME=\"$REPO_GIT_HASH\"" >> "$BUILD_METADATA_ENV_FILE"
```

Comment:
Interestingly, this will include the quotes in the output.

Reply:
Just looking at the original code I think that might have been what was intended. After all, otherwise the final quoted section is just an empty string. I'm guessing the intention might have been to use single quotes?